setup.py

```diff
@@ -161,7 +161,7 @@ def build_extension(self, ext):
     },
     ext_modules=[
         CMakeExtension("torch_mlir._mlir_libs._jit_ir_importer"),
-    ] if not TORCH_MLIR_ENABLE_ONLY_MLIR_PYTHON_BINDINGS else [CMakeExtension("_")],
+    ] if not TORCH_MLIR_ENABLE_ONLY_MLIR_PYTHON_BINDINGS else [CMakeExtension("torch_mlir._mlir_libs._torchMlir")],
```

Comment:
After much trial and tribulation, it was discovered that this is the correct way to do this (despite my comments to the contrary in the [previous PR](https://github.com/llvm/torch-mlir/pull/1854#discussion_r1103392622)). For anyone that cares: `setuptools` checks whether the package needs to be built [here](https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/command/build_ext.py#L517) and so `ext_path` needs to be the path to the extension as built by our CMake. With an empty or nonsense path it'll try to build the extension by passing sources to `cc` ([here](https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/command/build_ext.py#L548)) but because we have set [`sources=[]`](https://github.com/llvm/torch-mlir/blob/d30af8772b99fd6c71ed20af4e759fc3395ca6d4/setup.py#L134) in our `CMakeExtension` it will fail (citing no sources to compile).